src/shop.ts

```diff
@@ -58,6 +62,39 @@ export class Shop {
       timestamp: new Date().getTime(),
     }
 
-    return await this.storage.store(order)
+    return await this.storage.store(order, this.configuration.publicKey)
+  }
+
+  // back-end
+
+  async list(privateKey: JsonWebKey): Promise<Order[]> {
+    return this.storage.list(privateKey)
+  }
+
+  // setup
+
+  static async generateCrypto(): Promise<ShopCrypto> {
```

Comment:
How about 
```
export generateShopCrypto = async (): Promise<ShopCrypto> => {...}
```
instead of providing `static` function? There is no need to have a dependency to `Shop` class.

Reply:
Guess that comes from my Java/OO background. Exporting a function, not a proper class feels wrong to me. Also because it's two methods.

But I was thinking: I'd like to create `class Backend extends Shop`. Would that be OK with you as a solution? This way, I could even pull the `list()` and `load()` methods up to that new class, making `Shop` super clean and reduced to the minimum it needs. I really like the idea! Will propose code. :)